File: web-sources/s8-web-front/carbide/objform/BooleanObjFormGetter.js


import { PrimtiveObjFormGetter } from '/s8-web-front/carbide/objform/PrimitiveObjFormGetter.js';

/**
 * 
 */
export class BooleanObjFormGetter extends PrimtiveObjFormGetter {

    constructor() {
        super(5);
    }

    createOutputNode() {

        /* <output> */
        this.outputNode = document.createElement("div");
        this.outputNode.classList.add("objform-output");
        this.fieldNode.appendChild(this.outputNode);
        this.outputNode.innerHTML = `<span>(value)</span>`;
        /* </output> */
    }


    S8_set_value(value){
        this.outputNode.innerHTML = `<span>${value?"TRUE":"FALSE"}</span>`;
    }
}

File: mod-sources/com.s8.web.front/com/s8/web/front/carbide/objform/BooleanObjFormGetter.java
package com.s8.web.front.carbide.objform;

import com.s8.api.objects.web.WebS8Session;

public class BooleanObjFormGetter extends PrimitiveObjFormGetter {

	public BooleanObjFormGetter(WebS8Session branch) {
		super(branch, "/s8-web-front/carbide/objform/BooleanObjFormGetter");
	}
	
	/**
	 * Set field value
	 * @param value
	 */
	public void setValue(boolean value) {
		vertex.fields().setBool8Field("value", value);
	}
	
}
